### PR TITLE
Save arbitrary dimension `AbstractArray`s to SQL

### PR DESCRIPTION
**Pull request by samuelfneumann**
Added the ability to save `AbstractArray`s of arbitrary dimensions to a
SQL table.

Added a return to the main experiment of `examples/experiment.jl` to
demonstrate this.

### CODE REVIEW

**Comment by mkschleg** [comment]
Idk why my comments aren't coming through in the code review. But for the toml file, we shouldn't change where the experiment is. You should be running the examples in the examples folder, not in the reproduce root directory.

**Review by mkschleg — CHANGES_REQUESTED** [review]
The toml file needs to be modified as mentioned.

**Comment by mkschleg** [comment]
Also, now that I'm looking at it, why were the values in this toml changed? They were chosen specifically to help me debug the exception handling.

File: examples/configs/arg_iter_config_sql.toml [hunk 1]
```diff
@@ -6,7 +6,7 @@ save_backend="mysql" # mysql only database backend supported
 database="test_iter" # for params and resutsl
 save_dir="test_iter_sql" # for exceptions, settings, and more!
 
-exp_file="experiment.jl"
+exp_file="examples/experiment.jl"
```
[the remark this answers is not in the document]
Comment (samuelfneumann) [reply examples/configs/arg_iter_config_sql.toml]: Ah, I was running these in the reproduce root directory. I've reverted the toml file.

**Comment by samuelfneumann** [comment]
I was running the examples in the wrong directory, I've reverted the file.

**Pull request merged**